tc/sgx/trusted_worker_manager/enclave/work_order_data_handler.cpp

```diff
@@ -67,10 +65,10 @@ namespace tcf {
 
         input_data_hash = GetJsonStr(object, "dataHash");
         if (!encrypted_input_data.empty()) {
-            DecryptInputData(encrypted_input_data, iv);
-
+            DecryptInputData(data_encryption_key, iv, encrypted_input_data);
             if (!input_data_hash.empty()) {
-                VerifyInputHash(workorder_data.decrypted_data, HexStringToBinary(input_data_hash));
+                VerifyInputHash(workorder_data.decrypted_data,
```

Comment:
If decrypted_data is empty and input_data_hash is not empty it's an error condition

Reply:
ok